src/browser/providers/tidal.js

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+
+    setLoopStatus(status) {
+        if (this.controls.repeat) this.controls.repeat.click();
+        else super.setLoopStatus(status);
+    }
+
+    setVolume(volume) {
+        super.setVolume(volume);
+        if (this.controls.volumeProgress) {
+            this.controls.volumeProgress.value = volume * 100;
+            this.controls.volumeProgress.style =
+                '--val:' + volume * 100 + '; --max:100; --min:0;';
+        }
+    }
+
+    getVolume() {
+        if (this.controls.volumeProgress) {
+            return parseInt(this.controls.volumeProgress.value, 10) / 100;
+        }
+        return super.getVolume();
+    }
+
+    setRate() {
+        // disabled
+    }
+}
+
+Playback = TidalPlayback;
+
+window.addEventListener('mpris2-setup', function() {
+    /*
+     * Since tidal is loaded dynamically, we need to wait until the playback controls are added
+     * to the DOM before we can get the elements.
+     *
+     * We create an observer to monitor the added nodes, and set the playback controls when the
+     * player is loaded.
+     *
+     */
+    const playerObserver = new MutationObserver(mutations => {
+        mutations.forEach(mutation => {
+            mutation.addedNodes.forEach(n => {
+                if (n.className && n.className.startsWith('footerPlayer')) {
+                    initPage()
+                    playerObserver.disconnect()
+                }
+            });
+        });
+    })
+
+      const initPage = () => {
+            const BUTTONS_CONTAINER = 'div[class*="playbackControls"]';
+
+            page.playback.controls = {
+                shuffle: document.querySelector(
+                    BUTTONS_CONTAINER + ' button:nth-child(1)'
+                ),
+                prev: document.querySelector(
+                    BUTTONS_CONTAINER + ' button:nth-child(2)'
+                ),
+                play: document.querySelector(
+                    BUTTONS_CONTAINER + ' button:nth-child(3)'
+                ),
+                next: document.querySelector(
+                    BUTTONS_CONTAINER + ' button:nth-child(4)'
+                ),
+                volumeProgress: document.querySelector(
+                    'div[class*="volumeSlider"] input'
+                )
+            };
+
+            // We also reset the repeat button, since this might change
+            const setRepeatElement = () => {
+                page.playback.controls.repeat = document.querySelector(
+                    BUTTONS_CONTAINER + ' button:nth-child(5)'
+                );
+                // We might not be playing from a playlist, in this case the repeat button is
+                // replaced with a block button, which we dont want do click.
+                if (
+                    page.playback.controls.repeat.className.startsWith(
+                        'blockButton'
+                    )
+                ) {
+                    page.playback.controls.repeat == null;
```

Comment:
One last thing, this should be an assignment, right?
```suggestion
                    page.playback.controls.repeat = null;
```